stdlib/2/_io.pyi

```diff
@@ -19,91 +19,149 @@ class _IOBase(BinaryIO):
+    def readlines(self, hint: int = ...) -> list[bytes]: ...
     def next(self) -> bytes: ...
-    def __iter__(self) -> Iterator[bytes]: ...
-    def __enter__(self) -> '_IOBase': ...
-    def __exit__(self, t: Optional[Type[BaseException]], value: Optional[BaseException],
-                 # TODO: traceback should be TracebackType but that's defined in types
-                 traceback: Optional[Any]) -> bool: ...
 
 class _BufferedIOBase(_IOBase):
-    def read1(self, n: int) -> str: ...
-    def read(self, n: int = ...) -> str: ...
+    def read1(self, size: int) -> bytes: ...
+    def read(self, size: int = ...) -> bytes: ...
     def readinto(self, buffer: bytearray) -> int: ...
-    def write(self, s: str) -> int: ...
+    def write(self, s: bytes) -> int: ...
     def detach(self) -> "_BufferedIOBase": ...
+    def __enter__(self) -> '_BufferedIOBase': ...
 
 class BufferedRWPair(_BufferedIOBase):
-    def peek(self, n: int = ...) -> str: ...
+    def __init__(self, reader: _RawIOBase, writer: _RawIOBase) -> None: ...
+    def peek(self, n: int = ...) -> bytes: ...
+    def __enter__(self) -> 'BufferedRWPair': ...
 
 class BufferedRandom(_BufferedIOBase):
+    mode = ...  # type: str
     name = ...  # type: str
     raw = ...  # type: _IOBase
-    mode = ...  # type: str
-    def peek(self, n: int = ...) -> str: ...
+    def __init__(self, raw: _IOBase,
+                 buffer_size: int = ...,
+                 max_buffer_size: int = ...) -> None: ...
+    def peek(self, n: int = ...) -> bytes: ...
+    def __enter__(self) -> 'BufferedRandom': ...
 
 class BufferedReader(_BufferedIOBase):
+    mode = ...  # type: str
     name = ...  # type: str
     raw = ...  # type: _IOBase
-    mode = ...  # type: str
-    def peek(self, n: int = ...) -> str: ...
+    def __init__(self, raw: _IOBase, buffer_size: int = ...) -> None: ...
+    def peek(self, n: int = ...) -> bytes: ...
+    def __enter__(self) -> 'BufferedReader': ...
 
 class BufferedWriter(_BufferedIOBase):
     name = ...  # type: str
     raw = ...  # type: _IOBase
     mode = ...  # type: str
+    def __init__(self, raw: _IOBase,
+                 buffer_size: int = ...,
+                 max_buffer_size: int = ...) -> None: ...
+    def __enter__(self) -> 'BufferedWriter': ...
 
 class BytesIO(_BufferedIOBase):
+    def __init__(self, initial_bytes: bytes = ...) -> None: ...
     def __setstate__(self, tuple) -> None: ...
     def __getstate__(self) -> tuple: ...
-    def getvalue(self) -> str: ...
+    def getvalue(self) -> bytes: ...
+    def write(self, s: bytes) -> int: ...
+    def writelines(self, lines: Iterable[bytes]) -> None: ...
+    def read1(self, size: int) -> bytes: ...
+    def __iter__(self) -> Iterator[bytes]: ...
+    def next(self) -> bytes: ...
+    def __enter__(self) -> 'BytesIO': ...
 
 class _RawIOBase(_IOBase):
     def readall(self) -> str: ...
-    def read(self, n: int = ...) -> str: ...
+    def read(self, size: int = ...) -> str: ...
```

Comment:
Done. The C code seems inconsistent with regards to `n` and `size`. I shouldn't have made these changes in bulk.

(The parameter name probably doesn't matter, since these methods don't allow keyword args. But even so.)